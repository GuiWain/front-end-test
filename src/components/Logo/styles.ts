import styled, { css } from 'styled-components'
import UseLink from 'next/link'
import UseImage from 'next/image'

import { LogoProps } from '.'

export const Link = styled(UseLink)``

export const Image = styled(UseImage)`
  max-width: 100%;
  object-fit: contain;
`

const wrapperModifiers = {
  small: () => css`
    width: 6rem;
  `,
  normal: () => css`
    width: 8rem;
  `,
  large: () => css`
    width: 10rem;
  `,
  fullWidth: () => css`
    width: 12rem;
  `
}

export const Wrapper = styled.div.attrs({
  'data-testid': 'Wrapper'
})<LogoProps>`
  ${({ size }) => css`
    ${!!size && wrapperModifiers[size]}
  `}
`

export const Content = styled.span`
  cursor: pointer;
  display: flex;
  justify-content: center;
  align-items: center;
`
