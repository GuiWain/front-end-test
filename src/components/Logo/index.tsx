import * as S from './styles'

export type LogoProps = {
  size?: 'small' | 'normal' | 'large' | 'fullWidth'
}

const Logo = ({ size = 'normal', ...props }: LogoProps) => (
  <S.Wrapper size={size} {...props}>
    <S.Link href="/">
      <S.Content>
        <S.Image
          src="/img/logo-lorem-ipsum.png"
          alt="Logo Lorem Ipsum"
          width="600px"
          height="512px"
        />
      </S.Content>
    </S.Link>
  </S.Wrapper>
)

export default Logo
