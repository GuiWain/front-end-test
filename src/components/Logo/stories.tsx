import { Story, Meta } from '@storybook/react/types-6-0'
import Logo, { LogoProps } from '.'

export default {
  title: 'Components/Logo',
  component: Logo,
  argTypes: {
    size: {
      control: {
        type: 'radio',
        options: ['small', 'normal', 'large', 'fullWidth']
      }
    }
  }
} as Meta

export const Default: Story<LogoProps> = (args) => <Logo {...args} />

Default.args = {
  size: 'normal'
}
