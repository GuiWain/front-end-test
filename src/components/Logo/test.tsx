import { render, screen } from 'utils/test-utils'

import Logo from '.'

describe('<Logo />', () => {
  it('should render a normal size', () => {
    const { container } = render(<Logo />)
    expect(screen.getByRole('img')).toHaveAttribute('alt')

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render a small size', () => {
    render(<Logo size="small" />)
    expect(screen.getByTestId('Wrapper')).toHaveStyle({
      width: '6rem'
    })
  })
  it('should render a normal size', () => {
    render(<Logo size="normal" />)
    expect(screen.getByTestId('Wrapper')).toHaveStyle({
      width: '8rem'
    })
  })
  it('should render a large size', () => {
    render(<Logo size="large" />)
    expect(screen.getByTestId('Wrapper')).toHaveStyle({
      width: '10rem'
    })
  })
  it('should render a FullWidth', () => {
    render(<Logo size="fullWidth" />)
    expect(screen.getByTestId('Wrapper')).toHaveStyle({
      width: '12rem'
    })
  })
})
