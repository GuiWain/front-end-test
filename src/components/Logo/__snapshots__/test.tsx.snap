// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> should render a normal size 1`] = `
.c2 {
  max-width: 100%;
  object-fit: contain;
}

.c0 {
  width: 8rem;
}

.c1 {
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
  class="c0"
  data-testid="Wrapper"
>
  <span
    class="c1"
  >
    <div
      style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
    >
      <div
        style="box-sizing: border-box; display: block; max-width: 100%;"
      >
        <img
          alt=""
          aria-hidden="true"
          src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNjAwIiBoZWlnaHQ9IjUxMiIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
          style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
        />
      </div>
      <img
        alt="Logo Lorem Ipsum"
        class="c2"
        data-nimg="intrinsic"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
      />
      <noscript />
    </div>
  </span>
</div>
`;
